d3a_interface/utils.py

```diff
@@ -427,3 +428,28 @@ def scenario_representation_traversal(sc_repr, parent=None):
             yield from scenario_representation_traversal(child, sc_repr)
 
     yield sc_repr, parent
+
+
+class HomeRepresentationUtils:
+    @staticmethod
+    def _is_home(representation):
+        home_devices = ["PV", "Storage", "Load", "MarketMaker"]
+        return all("type" in c and c["type"] in home_devices
+                   for c in representation["children"])
+
+    @classmethod
+    def is_home_area(cls, representation):
+        is_market_area = not key_in_dict_and_not_none(representation, "type") or representation["type"] == "Area"
```

Comment:
What if the 'type' key doesn't exists ? this will pass the first check and then it will raise an exception in the `representation["type"] == "Area"`
```suggestion
        is_market_area = representation.get("type", True) == "Area"
```

Reply:
Actually it is working as it is, have a look at the unit tests where an Area type is not explicitly defined. The case of "type" key missing is covered by the `not key_in_dict_and_not_none(representation, "type")` statement, which will immediately return True if there is no type key.